ipaserver/plugins/idviews.py

```diff
@@ -715,6 +715,36 @@ def remove_ipaobject_overrides(ldap, api, dn):
             ldap.delete_entry(entry)
 
 
+# Extended user validator that allows to resolve users from trusted domains
+@member_validator(membertype='user')
+def validate_external_user(ldap, dn, keys, options, uid):
+    param = api.Object['user'].primary_key
+
+    try:
+        uid = param(uid)
+        param.validate(uid)
```

Comment:
It's not obvious what value ``uid`` is going to have in the except block. The code flow can lead to different values of ``uid`` in ``resolve_object_to_anchor()`` depending on the outcome of ``param(uid)``. I recommend to move ``uid = param(uid)`` out of the try/except block to make it obvious that the try/execpt block only handles the validate call.

Reply:
I moved it out (will push once temp commit passes/fails) but this is just the same code as in the `add_external_pre_callback()` before the change.